pkg/s3signer/request-signature-v2.go

```diff
@@ -316,7 +316,7 @@ func writeCanonicalizedResource(buf *bytes.Buffer, req http.Request, isPreSign b
 				// Request parameters
 				if len(vv[0]) > 0 {
 					buf.WriteByte('=')
-					buf.WriteString(strings.Replace(url.QueryEscape(vv[0]), "+", "%20", -1))
```

Comment:
@vadmeste can we test this with `ListObjects()` with some UTF-8 files and see how it works.